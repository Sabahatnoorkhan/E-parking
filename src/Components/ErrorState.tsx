import React from "react";
import { Container, Typography, Button, Box } from "@mui/material";
import ErrorIcon from '@mui/icons-material/Error';


const ErrorPage = () => {
  const handleRetry = () => {
    window.location.reload();
  };

  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
        height: "100vh",
        backgroundColor: "#f5f5f5",
      }}
    >
      <Container maxWidth="sm" sx={{ textAlign: "center" }}>
      <ErrorIcon
        sx={{
          fontSize: 100,
          color: 'error.main',
          marginBottom: '20px',
        }}
      />
        <Typography variant="h4" gutterBottom>
          Oops! Something went wrong.
        </Typography>
        <Typography variant="body1" color="textSecondary" paragraph>
          An error occurred while processing your request. Please try again.
        </Typography>
        <Button variant="contained" color="primary" onClick={handleRetry}>
          Retry
        </Button>
      </Container>
    </Box>
  );
};

export default ErrorPage;
