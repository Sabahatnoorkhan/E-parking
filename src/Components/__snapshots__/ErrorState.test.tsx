import React from 'react';
import { render } from '@testing-library/react';
import ErrorState from '../ErrorState';

describe('ErrorState Component', () => {
  it('Should match the snapshot', () => {
    const { asFragment } = render(<ErrorState errorHandler={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});