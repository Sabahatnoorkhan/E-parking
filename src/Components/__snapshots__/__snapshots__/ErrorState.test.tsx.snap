// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState Component Should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-l36efw"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm css-1tnit8n-MuiContainer-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-sc7i6q-MuiSvgIcon-root"
        data-testid="ErrorIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-2h2zm0-4h-2V7h2z"
        />
      </svg>
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Oops! Something went wrong.
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-120mgm7-MuiTypography-root"
      >
        An error occurred while processing your request. Please try again.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Retry
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
